protocol/messenger_communities.go

```diff
@@ -1778,3 +1782,107 @@ func (m *Messenger) SyncCommunitySettings(ctx context.Context, settings *communi
 	chat.LastClockValue = clock
 	return m.saveChat(chat)
 }
+
+func (m *Messenger) ExtractDiscordDataFromImportFiles(filesToImport []string) (*discord.ExtractedData, map[string]*discord.ImportError) {
+
+	extractedData := &discord.ExtractedData{
+		Categories:             map[string]*discord.Category{},
+		ExportedData:           make([]*discord.ExportedData, 0),
+		OldestMessageTimestamp: 0,
+	}
+
+	errors := map[string]*discord.ImportError{}
+
+	for _, fileToImport := range filesToImport {
+		filePath := strings.Replace(fileToImport, "file://", "", -1)
+		bytes, err := os.ReadFile(filePath)
+		if err != nil {
+			errors[fileToImport] = &discord.ImportError{
+				Code:    201,
+				Message: err.Error(),
+			}
+			continue
+		}
+
+		var discordExportedData discord.ExportedData
+
+		err = json.Unmarshal(bytes, &discordExportedData)
+		if err != nil {
+			errors[fileToImport] = &discord.ImportError{
+				Code:    202,
+				Message: err.Error(),
+			}
+			continue
+		}
+
+		if len(discordExportedData.Messages) == 0 {
+			errors[fileToImport] = &discord.ImportError{
+				Code:    203,
+				Message: "No messages to import",
+			}
+			continue
+		}
+
+		discordExportedData.Channel.FilePath = filePath
+		categoryID := discordExportedData.Channel.CategoryID
+
+		discordCategory := discord.Category{
+			ID:   categoryID,
+			Name: discordExportedData.Channel.CategoryName,
+		}
+
+		_, ok := extractedData.Categories[categoryID]
+		if !ok {
+			extractedData.Categories[categoryID] = &discordCategory
+		}
+
+		extractedData.ExportedData = append(extractedData.ExportedData, &discordExportedData)
+
+		if len(discordExportedData.Messages) > 0 {
+			layout := "2006-01-02T15:04:05+00:00"
+			msgTime, err := time.Parse(layout, discordExportedData.Messages[0].Timestamp)
+			if err != nil {
+				m.logger.Error("failed to parse discord message timestamp", zap.Error(err))
+				continue
+			}
+
+			if extractedData.OldestMessageTimestamp == 0 || int(msgTime.Unix()) <= extractedData.OldestMessageTimestamp {
+				// Exported discord channel data already comes with `messages` being
+				// sorted, starting with the oldest, so we can safely rely on the first
+				// message
+				extractedData.OldestMessageTimestamp = int(msgTime.Unix())
+			}
+		}
+	}
+	return extractedData, errors
+}
+
+func (m *Messenger) ExtractDiscordChannelsAndCategories(filesToImport []string) (*MessengerResponse, map[string]*discord.ImportError) {
```

Comment:
Because the underlying `ExtractDiscordDataFromImportFiles` API reutrns a `map[string]*discord.ImportError`, this API will do so as well, which is then also bubbles up to the RPC API.

As discussed offline, this works and should be fine. If you have any more thoughts or preferences here in terms of changes, please let me know